Assets/SteamVR_Unity_Toolkit/Scripts/Controls/3D/Utilities/VRTK_ContentHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿namespace VRTK
+{
+    using UnityEngine;
+
+    /// <summary>
+    /// This script will make objects inside of control contents properly interactable by reparenting them. 
+    /// Otherwise they would disappear if e.g. a drawer is closed even if the object was outside the drawer.
+    /// We need to go through collisions instead of hooking to the Ungrabbed event since ungrabbing will most likely happen above the inside causing the object to drop first. 
+    /// This approach is much more stable and allows for many more scenarios.
+    /// Not supported yet: detect if an object is taken from one content manager and dropped into another one.
+    /// </summary>
+    public class VRTK_ContentHandler : MonoBehaviour
+    {
+        public VRTK_Control control;
+        public Transform inside;
+        public Transform outside;
+
+        void Start()
+        {
+            VRTK_InteractableObject io = GetComponent<VRTK_InteractableObject>();
+            if (io == null)
+            {
+                // treat as parent and assign to all children
+                foreach (VRTK_InteractableObject childIo in GetComponentsInChildren<VRTK_InteractableObject>())
+                {
+                    if (childIo.GetComponent<VRTK_ContentHandler>() == null)
+                    {
+                        VRTK_ContentHandler ch = childIo.gameObject.AddComponent<VRTK_ContentHandler>();
+                        ch.control = control;
+                        ch.inside = inside;
+                        ch.outside = outside;
+                    }
+                }
+            }
+        }
+
+        public void OnCollisionEnter(Collision collision)
```

Comment:
According to the C# docs on msdn https://msdn.microsoft.com/en-us/library/ba0a1yw2.aspx (and this video from Unity https://youtu.be/XKGT_4mR1C4?t=163)

Any method that omits the access modifier is private by default. But they even (in that video) suggest to specify private/public if you have a mixture of access modifiers in the class.

That would lead me to believe the reason they're created without an access modifier by default (e.g. you get a `void Start()` and a void `Update()`) is because they are the only methods in that class (and therefore they seem to think omitting the access modifier is ok because it's assumed it's all private).

From this I would assume that the contrib guidelines are adhering to what they're suggesting. I wonder if anyone else has any ideas/understandings about this?